### PR TITLE
Moved the Menus out of the _SCENE_

### PR DESCRIPTION
**Pull request by dustinwloring1988**
Moved all the menus out of the scene only a few game objects let in the _SCENE_ now.

When all of the menus are loaded at run time from the resource folder, will parent to the UIMenus game object in the _SCENE_.

### CODE REVIEW

File: Assets/UIMenuController.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    {
+        GameObject tempGoObj;
+
+        tempGoObj = (GameObject)Instantiate(Resources.Load("UI/" + menuName));
+        tempGoObj.name = menuName;
+
+        tempGoObj.transform.SetParent(parent,false);
+
+        //Transform childTransform = child.transform;
+        //childTransform.parent = parent;
```
Comment (Randshot) [inline Assets/UIMenuController.cs]: Avoid pushing commented out code.

**Comment by Randshot** [comment]
The game runs just fine without any errors, but one of the UI elements is placed incorrectly.
placement

The code looks mostly clean and it goes without saying that there are many changes to the SCENE (duh).
The rest consists of the new prefabs.

**Comment by dustinwloring1988** [comment]
@Randshot - Fixed and Thank you

**Comment by vogonistic** [comment]
@dustinwloring1988 Get 2 people to sign off that everything works and I'll merge it. They have to verify that the menus and buttons still work as they should.

**Comment by longtomjr** [comment]
Starting testing now.

**Comment by Randshot** [comment]
I will check again after he updated the PR.

File: Assets/UIMenuController.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class UIMenuController : MonoBehaviour {
+
+    // This is the parent of the menus
+    Transform parent;
```
Comment (Grenkin1988) [inline Assets/UIMenuController.cs]: Access modifiers should be explicitly specified (in all other places also)

File: Assets/UIMenuController.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        AddMenu("MenuConstruction");
+	}
+
+    void Start()
+    {
+        AddMenu("MenuRight");
+    }
+
+	// Update is called once per frame
+	void Update ()
```
Comment (Grenkin1988) [inline Assets/UIMenuController.cs]: If it is empty, this could be removed

**Comment by dustinwloring1988** [comment]
Sorry guys my computer crashed just got it up and running again.

**Comment by dustinwloring1988** [comment]
@Randshot , @longtomjr   its update please try it again
Thanks

**Comment by Randshot** [comment]
@dustinwloring1988 Thanks. I am testing it right now and I am getting a weird error, so I am looking if it originates from this PR.

**Comment by dustinwloring1988** [comment]
@Randshot - Thanks for testing it out.. Need one more now

**Comment by abackwood** [comment]
Work on my system
:+1:

**Comment by Dormanil** [comment]
Looks good to me as well.

**Comment by dustinwloring1988** [comment]
Thanks @Dormanil @abackwood @Randshot

**Comment by vogonistic** [comment]
Merged! Reviewed by @Dormanil @abackwood @Randshot

**Pull request merged**